src/object.rs

```diff
@@ -58,7 +58,6 @@ pub struct Object {
     handle: CK_OBJECT_HANDLE,
     session: CK_SESSION_HANDLE,
     attributes: Vec<Attribute>,
-    modified: bool,
```

Comment:
Does rust have something like `gcc`s `-Werror` to treat warnings as an error. I think this could be handy for the CI to catch these issues in the future.

Reply:
I am not sure, but we could add a grep in CI and fail the test if we really want to, but there is still a couple of warning that *will* be fixed soonish and I do not want to force ourselves to remove and readd code just because of them.
Also in FIPS mode some things end up being unused but I do not want to have to add [cfg(not(feature = "fips"))] around things like interfaces in mechanism.rs